packages/firestore/src/api/user_data_converter.ts

```diff
@@ -242,6 +242,16 @@ class ParseContext {
     );
   }
 
+  /** Returns 'true' if 'fieldPath' was traversed when creating this context. */
+  contains(fieldPath: FieldPath): boolean {
```

Comment:
I think contains() can be misinterpreted to think the specified path must be contained within one of the writes in the ParseContext, but really it's the opposite (the specified path must contain one of the writes).  Maybe affects(), modifies(), or containsWriteBelow() ?  That said, I guess this is consistent with the existing ObjectValue.contains() so you can leave it as-is if you prefer.

Reply:
The function returns true if the original object that was being parsed contains a value at `fieldPath`. it is not necessarily a 1:1 match with writes, since a write at `a.b.c` contains `a.b`. The function returns false for an object `{a : b: {}}` and field path `a.b.c` (which I think is your second example). `contains` matches the semantics of `ObjectValue.contains()` and hence I'd prefer to keep the name.